pmm-app/src/pmm-qan-app-panel/QueryAnalytics/Filters/Filters.tsx

```diff
@@ -0,0 +1,174 @@
+        <div className={'filter-label'}>
+          <span className={'filter-name'}>
+            <CheckboxField
+              form={form}
+              name={`${group}:${item.value}`}
+              label={item.value}
+              checked={labels && labels[group] && labels[group].includes(item.value)}
+            />
+          </span>
+          <span className={'percentage'}>
+            <span>{Humanize.transform(item.main_metric_percent, 'percent')}</span>
+          </span>
+        </div>
+      );
+    });
+  return itemsList.length ? (
+    <div>
+      <Divider>{name}</Divider>
+      {itemsList}
+    </div>
+  ) : null;
+};
+
+const FILTERS_GROUPS = [
+  {
+    name: 'Environment',
+    dataKey: 'environment',
+  },
+  {
+    name: 'Cluster',
+    dataKey: 'cluster',
+  },
+  {
+    name: 'Replication Set',
+    dataKey: 'replication_set',
+  },
+  {
+    name: 'Database',
+    dataKey: 'database',
+  },
+  {
+    name: 'Schema',
+    dataKey: 'schema',
+  },
+  {
+    name: 'Node Name',
+    dataKey: 'node_name',
+  },
+  {
+    name: 'Service Name',
+    dataKey: 'service_name',
+  },
+  {
+    name: 'Client Host',
+    dataKey: 'client_host',
+  },
+  {
+    name: 'User Name',
+    dataKey: 'username',
+  },
+  {
+    name: 'Service Type',
+    dataKey: 'service_type',
+  },
+  {
+    name: 'Node Type',
+    dataKey: 'node_type',
+  },
+  {
+    name: 'City',
+    dataKey: 'city',
+  },
+  {
+    name: 'Availability Zone',
+    dataKey: 'az',
+  },
+];
+
+const Filters = () => {
+  const [showAll, showSetAll] = useState(true);
+  const [filter, setFilter] = useState('');
+  const [filters, setFilters] = useState({});
+  const [groups, setGroups] = useState([]);
+  const context = useContext(StateContext);
+
+  useEffect(() => {
+    (async () => {
+      const result = await FiltersService.getQueryOverviewFiltersList(context.state.labels || {}, context.state.from, context.state.to);
+      setFilters(result);
+      setGroups(FILTERS_GROUPS);
+    })();
+  }, [context.state.labels, context.state.from, context.state.to]);
+  return (
+    <FormFinal
+      onSubmit={() => {}}
+      render={(): ReactElement => {
+        const { form, handleSubmit } = useForm({
+          onSubmit: filters => {},
+          validate: () => {},
+          initialValues: {},
+        });
+        // @ts-ignore
+        return (
+          <form
+            onSubmit={handleSubmit}
+            className="add-instance-form app-theme-dark"
+            onChange={() => {
+              context.dispatch({
+                type: 'SET_LABELS',
+                payload: { labels: form.getState().values },
+              });
+            }}
+          >
+            <div className={'filters-header'} style={{ padding: '5px 0px', height: '50px' }}>
+              <h5 style={{ margin: '3px', marginRight: '15px' }}>Filters</h5>
+              {showAll ? (
+                <a href="#" className={'filter-switchers'} onClick={showSetAll.bind(null, false)}>
```

Comment:
there should be no `.bind` calls inside components, pls prefer arrow functions instead